core/new-gui/src/app/workspace/service/workflow-graph/model/shared-model-change-handler.ts

```diff
@@ -301,38 +303,70 @@ export class SharedModelChangeHandler {
-            const addedOutputPort = event.delta[1]?.insert ? event.delta[1]?.insert : event.delta[0]?.insert;
-            if (addedOutputPort) {
-              this.onOperatorPortAdded(
-                operatorID,
-                false,
-                (addedOutputPort as Y.Map<any>[])[0].toJSON() as PortDescription
-              );
-            }
-            if (event.delta[0]?.delete || event.delta[1]?.delete) {
-              this.onOperatorPortRemoved(operatorID, false);
-            }
           } else if (event.path.includes("operatorProperties")) {
             this.onOperatorPropertyChanged(operatorID, event.transaction.local);
+          } else if (event.path.includes("inputPorts")) {
+            this.handlePortEvent(event, operatorID, true);
+          } else if (event.path.includes("outputPorts")) {
+            this.handlePortEvent(event, operatorID, false);
+          } else {
+            throw new Error(`undefined operation on shared type: .${event}`);
           }
         }
       });
     });
   }
 
+  /**
+   * Handles the additon, deletion and deeper changes to operator ports.
+   * @param event
+   * @param operatorID
+   * @param isInput Since input and output ports are separate properties, need to access them differently.
+   * @private
+   */
+  private handlePortEvent(event: Y.YEvent<Y.Map<any>>, operatorID: string, isInput: boolean) {
+    if (event.path.length === 2) {
+      // Port added or deleted
+      const addedPort = event.delta[1]?.insert ? event.delta[1]?.insert : event.delta[0]?.insert;
+      if (addedPort) {
+        this.onPortAdded(operatorID, isInput, (addedPort as Y.Map<any>[])[0].toJSON() as PortDescription);
+      }
+      if (event.delta[0]?.delete || event.delta[1]?.delete) {
+        this.onPortRemoved(operatorID, isInput);
+      }
+    } else {
+      const changedOperator = this.texeraGraph.getOperator(operatorID);
+      if (event.path.includes("displayName")) {
+        // Display name changed (via shared text editor)
+        const changedPort = isInput
+          ? changedOperator.inputPorts[event.path[2] as number]
+          : changedOperator.outputPorts[event.path[2] as number];
+        this.texeraGraph.portDisplayNameChangedSubject.next({
+          operatorID: operatorID,
+          portID: changedPort.portID,
+          newDisplayName: event.target.toJSON() as unknown as string,
+        });
+      } else if (event.path.length >= 3) {
+        // Port property changed
+        const newPortDescription = isInput
+          ? changedOperator.inputPorts[event.path[2] as number]
+          : changedOperator.outputPorts[event.path[2] as number];
+        this.texeraGraph.portPropertyChangedSubject.next({
+          operatorPortID: {
+            operatorID: operatorID,
+            portID: newPortDescription.portID,
+          },
+          newProperty: {
+            partitionInfo: newPortDescription.partitionRequirement as PartitionInfo,
```

Comment:
it's better to align the names/types than perform conversions, otherwise, this code can be confusing to maintain.

Reply:
Here the conversion is done mainly because the `partitionRequirement` is an optional field of `PortDescription`, since it is not required for a port. But for this event subject specifically it is not optional, and we know the two fields are not `undefined` (but eslint does not know, so if we do not do this conversion it will throw errors.)

Reply:
is `partitionRequirement` and `PartitionInfo` the same thing? My point is we'd better use the same name if they are actually the same thing.